python/ray/tests/test_failure.py

```diff
@@ -1245,14 +1245,11 @@ def test_node_failure(node_to_kill, use_actors):
         test_process_failure(use_actors)
 
     ray.state.state._check_connected()
-    keys = [
-        key for r in ray.state.state.redis_clients
-        for key in r.keys("WORKER_FAILURE*")
-    ]
+    dead_workers = ray.state.state.dead_workers()
     if node_failure:
-        assert len(keys) <= 1, len(keys)
```

Comment:
cc @stephanie-wang do you know why the assertion is `assert len(keys) <= 1`? It looks like we removed Worker_Failure channel to report failed worker info, and it succeeds because the key length was always 0. Also, isn't this supposed to be `len(keys) == 1` instead?